app/assets/javascripts/cookie_functions.js

```diff
@@ -1,22 +1,35 @@
 "use strict";
 
-window.TeacherPayments = window.TeacherPayments || {};
-
-window.TeacherPayments.cookies = {
-  acceptCookieName: "accept_cookies",
-  checkNonEssentialCookiesAccepted: function() {
-    return this._get(this.acceptCookieName);
-  },
-  acceptNonEssentialCookies: function() {
-    this._set(this.acceptCookieName, 1, 90);
-  },
-  _set: function(name, value, days) {
+(function() {
+  window.TeacherPayments = window.TeacherPayments || {};
+  window.TeacherPayments.cookies = window.TeacherPayments.cookies || {};
+  window.TeacherPayments.cookies.postNonEssentialCookieAcceptanceFunctions =
+    window.TeacherPayments.cookies.postNonEssentialCookieAcceptanceFunctions || [];
+
+  var acceptCookieName = "accept_cookies";
+
+  function setCookie(name, value, days) {
     var date = new Date();
     date.setTime(date.getTime() + 24 * 60 * 60 * 1000 * days);
     document.cookie = name + "=" + value + ";path=/;expires=" + date.toGMTString();
-  },
-  _get: function(name) {
+  }
+
+  function getCookie(name) {
     var v = document.cookie.match("(^|;) ?" + name + "=([^;]*)(;|$)");
     return v ? v[2] : null;
   }
-};
+
+  window.TeacherPayments.cookies.checkNonEssentialCookiesAccepted = function checkNonEssentialCookiesAccepted() {
```

Comment:
Do we need to repeat the function name here? Could we do `window.TeacherPayments.cookies.checkNonEssentialCookiesAccepted = function() {`

Reply:
It helps with debugging. If there's an error, it will be in the anonymous function, while if the function is named, we get the function name back. This may have improved since the last time I had to think about it, though. It's not as much of a consideration when writing modern JS.

Reply:
👍